ocs_ci/ocs/platform_nodes.py

```diff
@@ -1685,6 +1686,67 @@ def read_default_config(self, default_config_path):
         return default_config_dict
 
 
+class IBMPowerNodes(BaremetalNodes):
+    """
+    IBM Power Nodes class
+    """
+
+    def ___init__(self):
+        super(IBMPowerNodes, self).__init__()
+        self.powernodes = powernodes.POWERNODES()
```

Comment:
I don't understand the need for the separate class here; I'd argue that all functionality belongs in one class (even if it should potentially be in a different module and imported into `power_nodes`)

Reply:
I see a pattern among other players in this code, they drive changes to OCP from platform_nodes and underlying code is from ocs_ci.utility. I wanted to keep the same structure for this new code as well.

Reply:
I see a pattern, changes to OCP infrastructure are driven from platform_nodes, but underlying changes are maintained in utility. I wanted to keep the same structure for this new code.

Reply:
> I don't understand the need for the separate class here; I'd argue that all functionality belongs in one class (even if it should potentially be in a different module and imported into `power_nodes`)

Zack, I am not sure I understand your comment.   There are multiple classes today and none of them support libvirt or powervs.   Power actually has two platforms - libvirt and powervs.   There is no standardization around libvirt as each node may have a different VM name.   Power follows this pattern test-<ocp version>-<role>, but another implementation may utilize a different naming scheme.  This is really specific to IBM.  There is no platform independent implementation for libvirt running ocs-ci today, just IBM.   That being said, this code would probably result in an exception on powervs as libvirt is not even configured.  The virsh command is probably not even installed.  So, I think the new class for power needs to import modules for ibm-libvirt and powervs.

Reply:
If this is just following existing patterns in the project, I'll just withdraw my comment.